### PR TITLE
Add better hash URL support.

### PR DESCRIPTION
**Pull request by arunoda**
This PR comes with following enhancements related to hash url changes.

1. Add scrolling to given id related to hash
2. Hash changes won't trigger getInitialProps

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -29,6 +29,24 @@ export default class App extends Component {
 }
 
 class Container extends Component {
+  componentDidMount () {
+    this.scrollToHash()
```
Comment (arunoda) [inline lib/app.js]: We need to do the scrolling here because, in the router there's no way we could know when the correct DOM is available.

**Comment by arunoda** [comment]
@rauchg take this now.

File: lib/app.js [hunk 2]
```diff
@@ -29,6 +29,24 @@ export default class App extends Component {
...
+  }
+
+  componentDidUpdate () {
+    this.scrollToHash()
+  }
+
+  scrollToHash () {
+    const { hash } = this.props
+    const el = document.getElementById(hash)
+    if (el) {
```
Comment (ericf) [inline lib/app.js]: What about `<a name="some-hash">`?

Comment (arunoda) [inline lib/app.js]: Oh! I didn't know it's supported. Let's add it.
PRs welcome always :)

**Pull request merged**